source/classification2.md

```diff
@@ -575,46 +685,68 @@ _ctab = pd.crosstab(cancer_test_predictions["Class"],
             cancer_test_predictions["predicted"]
            )
 
-glue("confu11", _ctab["Malignant"]["Malignant"])
-glue("confu00", _ctab["Benign"]["Benign"])
-glue("confu10", _ctab["Benign"]["Malignant"])
-glue("confu01", _ctab["Malignant"]["Benign"])
-glue("confu11_00", _ctab["Malignant"]["Malignant"] + _ctab["Benign"]["Benign"])
-glue("confu10_11", _ctab["Benign"]["Malignant"] + _ctab["Malignant"]["Malignant"])
-glue("confu_fal_neg", round(100 * _ctab["Benign"]["Malignant"] / (_ctab["Benign"]["Malignant"] + _ctab["Malignant"]["Malignant"])))
+c11 = _ctab["Malignant"]["Malignant"]
+c00 = _ctab["Benign"]["Benign"]
+c10 = _ctab["Benign"]["Malignant"] # classify benign, true malignant 
+c01 = _ctab["Malignant"]["Benign"] # classify malignant, true benign
+
+glue("confu11", c11)
+glue("confu00", c00)
+glue("confu10", c10)
+glue("confu01", c01)
+glue("confu11_00", c11 + c00)
+glue("confu10_11", c10 + c11)
+glue("confu_fal_neg", np.round(100 * c10 / (c10 + c11)))
+glue("confu_accuracy", np.round(100*(c00+c11)/(c00+c11+c01+c10),2))
+glue("confu_precision", np.round(100*c11/(c11+c01), 2))
+glue("confu_recall", np.round(100*c11/(c11+c10),2))
+glue("confu_precision_0", np.round(100*c11/(c11+c01)))
+glue("confu_recall_0", np.round(100*c11/(c11+c10)))
 ```
 
 The confusion matrix shows {glue:}`confu11` observations were correctly predicted 
-as malignant, and {glue:}`confu00` were correctly predicted as benign. Therefore the classifier labeled 
-{glue:}`confu11` + {glue:}`confu00` = {glue:}`confu11_00` observations 
-correctly. It also shows that the classifier made some mistakes; in particular,
+as malignant, and {glue:}`confu00` were correctly predicted as benign. 
+It also shows that the classifier made some mistakes; in particular,
 it classified {glue:}`confu10` observations as benign when they were truly malignant,
 and {glue:}`confu01` observations as malignant when they were truly benign.
+Using our formulas from earlier, we see that the accuracy agrees with what Python reported,
+and can also compute the precision and recall of the classifier:
+
+$$\mathrm{accuracy} = \frac{\mathrm{number \; of  \; correct  \; predictions}}{\mathrm{total \;  number \;  of  \; predictions}} = \frac{{glue:}`confu00`+{glue:}`confu11`}{{glue:}`confu00`+{glue:}`confu11`+{glue:}`confu01`+{glue:}`confu10`} = {glue:}`confu_accuracy`$$
+
```

Comment:
@joelostblom -- this glue code doesn't work in latex. I couldn't figure out how to glue code results into latex equations in jupyterbook. Any idea how to do this / have you run into that before?

Reply:
nevermind, got it -- using `from IPython.display import Math`, constructing a math expression manually as a python string, gluing it, and then rendering the glue via `glue:math`

Reply:
Great that you got it working, I don't know a better way on the top of my head